artiq_env.nix

```diff
@@ -0,0 +1,31 @@
+let
+  # Contains the NixOS package collection. ARTIQ depends on some of them, and
+  # you may also want certain packages from there.
+  pkgs = import <nixpkgs> {};
+  artiq-full = import <artiq-full> { inherit pkgs; };
+in
+  pkgs.mkShell {
+    buildInputs = [
+      (pkgs.python3.withPackages(ps: [
+        # List desired Python packages here.
+        artiq-full.artiq
+        artiq-full.artiq-comtools
+        # The board packages are also "Python" packages. You only need a board
+        # package if you intend to reflash that board (those packages contain
+        # only board firmware).
+        artiq-full.artiq-board-kc705-nist_clock
+        artiq-full.artiq-board-kasli-wipm
+        # from the NixOS package collection:
+        ps.paramiko  # needed for flashing boards remotely (artiq_flash -H)
+        ps.pandas
```

Comment:
I'd like to keep this image small. Can we drop pandas, numba, bokeh, spyder in this image and add them to a superset image? If docker compose and nix play together this would be simple.

Reply:
Oops, apologies: this PR was meant to be on my own personal repo, I wasn't intending to submit it to yours yet! I'll close this PR, and maybe resubmit once it's in a better shape